### PR TITLE
Need later version & better way to locate cuFile lib

### PR DESCRIPTION
**Pull request by qkoziol**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -32,20 +32,20 @@ project(HDF5_VFD_GDS C)
 # Setup install and output Directories
 #------------------------------------------------------------------------------
 if(NOT HDF5_VFD_GDS_INSTALL_BIN_DIR)
```
Comment (Jacobfaib) [inline CMakeLists.txt:36]:
```suggestion
include(GNUInstallDirs)

if(NOT HDF5_VFD_GDS_INSTALL_BIN_DIR)
```

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -45,6 +45,7 @@ target_link_libraries(hdf5_vfd_gds
   ${HDF5_VFD_GDS_EXPORTED_LIBS}
```
Comment (Jacobfaib) [inline src/CMakeLists.txt:46]:
```suggestion
  PUBLIC
  ${HDF5_VFD_GDS_EXPORTED_LIBS}
```
Need to label the others in order to use PRIVATE. Assuming these are PUBLIC? Be good if someone could clarify.

File: CMakeLists.txt [hunk 3]
```diff
@@ -69,18 +69,9 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${HDF5_VFD_GDS_CMAKE_MODULE_PATH})
...
-find_library (HDF5_VFD_GDS_CUFILE_LIB
-  NAMES
-    cufile
-  HINTS
-    "${HDF5_VFD_GDS_CUFILE_DIR}/lib"
-    "${HDF5_VFD_GDS_CUFILE_DIR}/lib64"
-  REQUIRED
-)
+find_package(CUDAToolKit REQUIRED)
+target_link_libraries(${HDF5_VFD_GDS_CUFILE_LIB} PRIVATE CUDA::cuFile)
```
Comment (Jacobfaib) [inline CMakeLists.txt]:
```suggestion
```

File: src/CMakeLists.txt (hunk 2 above)
Comment (qkoziol) [reply src/CMakeLists.txt:46]: Yes, those should be public

File: CMakeLists.txt (hunk 3 above)
Comment (qkoziol) [reply CMakeLists.txt]: Missed than, thanks!

File: CMakeLists.txt [hunk 4]
```diff
@@ -69,18 +71,8 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${HDF5_VFD_GDS_CMAKE_MODULE_PATH})
 #------------------------------------------------------------------------------
 find_package (CUDA REQUIRED)
```
Comment (Jacobfaib) [inline CMakeLists.txt]:
```suggestion
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -69,18 +71,8 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${HDF5_VFD_GDS_CMAKE_MODULE_PATH})
...
-    cufile
-  HINTS
-    "${HDF5_VFD_GDS_CUFILE_DIR}/lib"
-    "${HDF5_VFD_GDS_CUFILE_DIR}/lib64"
-  REQUIRED
-)
+find_package(CUDAToolKit REQUIRED)
 
 set(HDF5_VFD_GDS_EXT_INCLUDE_DEPENDENCIES
   ${HDF5_VFD_GDS_EXT_INCLUDE_DEPENDENCIES}
```
Comment (Jacobfaib) [inline CMakeLists.txt:77]: Github doesn't let me add suggestions to the lines below, but swap these:
```diff
- CUDA_INCLUDE_DIRS
+ CUDAToolkit_INCLUDE_DIRS
```
and get rid of these variables
```diff
- CUDA_LIBRARIES
- HDF5_VFD_GDS_CUFILE_LIB
```

File: src/CMakeLists.txt [hunk 6]
```diff
@@ -42,9 +42,11 @@ target_include_directories(hdf5_vfd_gds
   SYSTEM PUBLIC ${HDF5_VFD_GDS_EXT_INCLUDE_DEPENDENCIES}
 )
 target_link_libraries(hdf5_vfd_gds
+  PUBLIC
   ${HDF5_VFD_GDS_EXPORTED_LIBS}
   ${HDF5_VFD_GDS_EXT_LIB_DEPENDENCIES}
   ${HDF5_VFD_GDS_EXT_PKG_DEPENDENCIES}
+  PRIVATE CUDA::cuFile
```
Comment (Jacobfaib) [inline src/CMakeLists.txt]:
```suggestion
  PRIVATE CUDA::cuFile CUDA::cudart_static
```

File: CMakeLists.txt (hunk 5 above)
Comment (qkoziol) [reply CMakeLists.txt:77]: Done, thanks!

**Review by hyoklee — APPROVED** [review]

File: CMakeLists.txt [hunk 7]
```diff
@@ -67,29 +69,16 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${HDF5_VFD_GDS_CMAKE_MODULE_PATH})
...
 # Try to locate cufile library
-find_library (HDF5_VFD_GDS_CUFILE_LIB
-  NAMES
-    cufile
-  HINTS
-    "${HDF5_VFD_GDS_CUFILE_DIR}/lib"
-    "${HDF5_VFD_GDS_CUFILE_DIR}/lib64"
-  REQUIRED
-)
+find_package(CUDAToolKit REQUIRED)
```
Comment (Jacobfaib) [inline CMakeLists.txt]:
```suggestion
find_package(CUDAToolkit REQUIRED)
```

**Review by hyoklee — APPROVED** [review]

File: CMakeLists.txt (hunk 7 above)
Comment (Jacobfaib) [reply CMakeLists.txt]: ping @qkoziol

**Pull request merged**